javascript/javascript3/week2/main/functions.js

```diff
@@ -0,0 +1,238 @@
+            "card_flipped_" +
+            amountOfCards +
+            " card_is-flipped card_changed-color card_not_faded";
+
+          //checking, if there are two cards opened
+          checkAmountOfCardsOpened(cardsOpened, deskCard, amountOfCards);
+        } else if (
+          deskCard.className ===
+          "card_flipped_" +
+            amountOfCards +
+            " card_is-flipped card_changed-color card_not_faded"
+        ) {
+          deskCard.removeAttribute("style");
+          deskCard.className =
+            "card_" + amountOfCards + " card_changed-color_2";
+          resetTheAmountOfCards();
+        } else if (
+          deskCard.className ===
+          "card_" + amountOfCards + " card_changed-color_2"
+        ) {
+          countFlipping();
+          deskCard.setAttribute(
+            "style",
+            'background-image: url("' + card.url + '");'
+          );
+          deskCard.className =
+            "card_flipped_" +
+            amountOfCards +
+            " card_is-flipped card_changed-color card_not_faded";
+
+          //checking, if there are two cards opened
+          checkAmountOfCardsOpened(cardsOpened, deskCard, amountOfCards);
+        }
+      });
+    });
+  });
+}
+
+function resetTheAmountOfCards() {
+  cardsOpened.length = 0;
+  console.log(cardsOpened);
+}
+
+//checking amount of cards, that were opened
+function checkAmountOfCardsOpened(cardsOpened, cardOnDesk, amountOfCards) {
+  if (cardsOpened.length === 1) {
+    cardsOpened.push(cardOnDesk.id);
+    setTimeout(function () {
+
+      //checking, if these two cards are same
+      if (cardsOpened[0] === cardsOpened[1]) {
+        if (amountOfCardsOnTheDesk > 2) {
+          amountOfCardsOnTheDesk -= 2;
+        } else {
+          endGame();
+        }
+        cardsOpened.length = 0;
+        console.log(cardsOpened);
+        const deskCardsToRemove = document.querySelectorAll(
+          ".card_flipped_" + amountOfCards
+        );
+        deskCardsToRemove.forEach(function (card) {
+          card.classList.add("card_fade");
+          setTimeout(function () {
+            deskMenu.removeChild(card);
+          }, 1000);
+        });
+      } else {
+        cardsOpened.length = 0;
+        console.log(cardsOpened);
+        const deskCardsToHide = document.querySelectorAll(
+          ".card_flipped_" + amountOfCards
+        );
+        deskCardsToHide.forEach(function (card) {
+          card.className = "card_" + amountOfCards;
+          card.removeAttribute("style");
+        });
+      }
+    }, 1000);
+    console.log(cardsOpened);
+  } else {
+    cardsOpened.push(cardOnDesk.id);
+    console.log(cardsOpened);
+  }
+}
+
+function pushingCardsToDesk(amountOfCards, cards) {
+
+  //pushing some amount of cards to desk-array and doubling them (that's why I'm pushing this cards two times)
+  for (let i = 0; i < amountOfCards; i++) {
+    desk.push(cards[i]);
+    desk.push(cards[i]);
+  }
+}
+
+function startTimer(count) {
```

Comment:
Do you think you could implement this function with only a single `setInterval`?